cvmfs/cvmfs_server

```diff
@@ -5077,6 +5063,176 @@ snapshot() {
+Suggested content:
+/var/log/cvmfs/*.log {
+    weekly
+    missingok
+    notifempty
+}
+EOF
+    exit 1
+  fi
+
+  if [ $separate_logs == 0 ]; then
+    # write into a temporary file in case more than one is active at the
+    #  same time
+    fulllog=/var/log/cvmfs/snapshots.log
+    log=/tmp/cvmfs_snapshots.$$.log
+    trap "rm -f $log" 0
+    (echo; echo "Logging in $log at `date`") >>$fulllog
+  fi
+
+  # Sort the active repositories by last snapshot time when on local storage.
+  # For non-local, swissknife only supports checking whether a file exists,
+  #  so only check whether non-initial snapshots are being skipped.
+  repos="$(for replica in /etc/cvmfs/repositories.d/*/replica.conf; do
+
+    # get repository information
+    local repodir="${replica%/*}"
+    repo="${repodir##*/}"
+    unset CVMFS_REPLICA_ACTIVE # remove previous setting, default is yes
+    load_repo_config $repo
+
+    if [ "$CVMFS_REPLICA_ACTIVE" = "no" ]; then
+      continue
+    fi
+
+    local upstream=$CVMFS_UPSTREAM_STORAGE
+    local snapshot_time=0
+    if is_local_upstream $upstream; then
+      local storage_dir=$(get_upstream_config $upstream)
+      local snapshot_file=$storage_dir/.cvmfs_last_snapshot
+      if [ -f $snapshot_file ]; then
+        snapshot_time="$(stat --format='%Y' $snapshot_file)"
+      elif [ $skip_noninitial == 1 ]; then
+        continue
+      fi
+    elif [ $skip_noninitial == 1 ]; then
+      if $user_shell "$(__swissknife_cmd) peek -d .cvmfs_last_snapshot -r ${upstream}" | grep -v -q "available"; then
+        continue
+      fi
+    fi
+
+    echo "${snapshot_time}:${repo}"
+
+  done|sort -n|cut -d: -f2)"
+
+  for repo in $repos; do 
+    if [ $separate_logs == 1 ]; then
+      log=/var/log/cvmfs/$repo.log
+    fi
+
+    (
+    echo
+    echo "Starting $repo at `date`"
+    if ! __do_snapshot $repo 1; then
+      echo "ERROR from cvmfs_server snapshot!" >&2
+    fi
+    echo "Finished $repo at `date`"
+    ) >> $log 2>&1
+
+    if [ $separate_logs == 0 ]; then
+      cat $log >>$fulllog
+      > $log
+    fi
+
+  done
+}
+
+snapshot() {
+  local alias_names
+  local retcode=0
+  local abort_on_conflict=0
+  local do_all=0
+  local allopts=""
+
+  OPTIND=1
+  while getopts "atsni" option; do
+    case $option in
+      a)
+        do_all=1
+      ;;
+      s|n|i)
+        allopts="$allopts -$option"
+      ;;
+      t)
+        abort_on_conflict=1
+      ;;
+      ?)
+        shift $(($OPTIND-2))
+        usage "Command snapshot: Unrecognized option: $1"
+      ;;
+    esac
+  done
+  shift $(($OPTIND-1))
+
+  if [ $do_all == 1 ]; then
+    [ $# == 0 ] || die "no non-option parameters expected with -a"
+
+    # ignore if there's a -t option, it's always implied with -a
+
+    __do_all_snapshots $allopts
+
+    # always return success because this is used from cron and we
+    #  don't want cron sending an email every time something fails
+    # errors will be in the log
```

Comment:
I don't like this approach.  If admins didn't want failure notifications, they can add:
```
MAILTO=
```
in the entry in `cron.d`.

I would propose if any repo fails, then print a message containing all failed repos and return non-zero.